sdmetrics/reports/single_table/base_report.py

```diff
@@ -0,0 +1,221 @@
+
+    def _get_num_iterations(self, property_name, metadata):
+        raise NotImplementedError
+
+    def generate(self, real_data, synthetic_data, metadata, verbose=True):
+        """Generate report.
+
+        Args:
+            real_data (pandas.DataFrame):
+                The real data.
+            synthetic_data (pandas.DataFrame):
+                The synthetic data.
+            metadata (dict):
+                The metadata, which contains each column's data type as well as relationships.
+            verbose (bool):
+                Whether or not to print report summary and progress.
+        """
+        self.validate(real_data, synthetic_data, metadata)
+
+        scores = []
+        progress_bar = None
+        if verbose:
+            sys.stdout.write('Generating report ...\n')
+
+        for ind, property_name in enumerate(self._properties):
+            if verbose:
+                num_iterations = self._get_num_iterations(property_name, metadata)
+                progress_bar = tqdm.tqdm(total=num_iterations, file=sys.stdout)
+                progress_bar.set_description(
+                    f'({ind + 1}/{len(self._properties)}) Evaluating {property_name}: '
+                )
+
+            score = self._properties[property_name].get_score(
+                real_data, synthetic_data, metadata, progress_bar=progress_bar
+            )
+            scores.append(score)
+            if verbose:
+                progress_bar.close()
+
+        self._overall_score = np.nanmean(scores)
+        self.is_generated = True
+
+        if verbose:
+            self._print_results(sys.stdout)
+
+    def _check_property_name(self, property_name):
+        """Check that the given property name is valid.
+
+        Args:
+            property_name (str):
+                The name of the property to check.
+        """
+        if property_name not in self._properties:
+            valid_property_names = "', '".join(self._properties.keys())
+            raise ValueError(
+                f"Invalid property name '{property_name}'."
+                f" Valid property names are '{valid_property_names}'."
+            )
+
+    def _check_report_generated(self):
+        if not self.is_generated:
+            raise ValueError('The report has not been generated. Please call `generate` first.')
+
+    def _validate_property_generated(self, property_name):
+        """Validate that the given property name and that the report has been generated."""
+        self._check_property_name(property_name)
+        self._check_report_generated()
+
+    def get_properties(self):
+        """Return the property score.
+
+        Returns:
+            pandas.DataFrame
+                The property score.
+        """
+        self._check_report_generated()
+        name, score = [], []
+        for property_name in self._properties:
+            name.append(property_name)
+            score.append(self._properties[property_name]._compute_average())
```

Comment:
you can iterate over `items` and there you will have the property and the property name:
```python
for property_name, property in self._properties.items():
    name.append(property_name)
    score.append(property._compute_average())
```